solutions/usaco-1061.mdx

```diff
@@ -24,6 +24,90 @@ Now we can sort cows by their static coordinates, which ensures that earlier col
 
 **Time Complexity:** $\mathcal{O}(N^2)$
 <LanguageSection>
+<CPPSection>
+	
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	int N; cin >> N;
+	vector<array<int, 3>> n_cows, e_cows;
+
+	for (int i = 0; i < N; i++) {
+		string dir;
+		int x, y;
+		cin >> dir >> x >> y;
+
+		// we need to add i as well because we need to output them in order
+		if (dir == "N") {
+			n_cows.push_back({x, y, i});
+		} else {
+			e_cows.push_back({x, y, i});
+		}
+	}
+
+	// sort north cows by x value
+	sort(n_cows.begin(), n_cows.end());
+
+	// sort east cows by y value
+	sort(e_cows.begin(), e_cows.end(), [](auto &left, auto &right) {
+		return left[1] < right[1];
+	});
+
+	/* 
+	 * we sort them by x/y value because we know that, for a single cow, 
+	 * it reaches a lower x/y value and then a higher one. 
+	 * Thus, it is more we would not miss any earlier collisions
+	 * if we iterate from lower to a higher value. 
+	 */
+
+	// stores the amount of grass it eats
+	vector<int> stop(N);
+	for (int i = 0; i < N; i++) {
+		stop[i] = -1;
+  	}
+
+
+	for (auto n_cow : n_cows) {
+    		for (auto e_cow : e_cows) {
```

Comment:
can you not use auto

Reply:
I don't think I can change it... `array` doesn't work for some reason

Reply:
k then